server/graphql/v1/mutations/collectives.js

```diff
@@ -177,6 +178,74 @@ export async function createCollective(_, args, req) {
   return collective;
 }
 
+export async function createCollectiveFromGithub(_, args, req) {
+  if (!req.remoteUser) {
+    throw new errors.Unauthorized({
+      message: 'You need to be logged in to create a collective',
+    });
+  }
+
+  if (!args.collective.name) {
+    throw new errors.ValidationFailed({ message: 'collective.name required' });
+  }
+
+  let collective;
+  const collectiveData = { ...args.collective };
+  const user = await models.User.findByPk(req.remoteUser.id);
+  if (!user) {
+    return Promise.reject(new Error(`No user with Id ${req.remoteUser.id} found`));
+  }
+
+  const existingCollective = await models.Collective.findOne({
+    where: { slug: collectiveData.slug.toLowerCase() },
+  });
+
+  if (existingCollective) {
```

Comment:
Maybe we should just fail and say that the collective slug is already taken. Let the user try again with another. Later, we could add some early validation in the frontend to avoid failing so late.

Reply:
Yeah, that is how it is currently done in `createCollective` resolver. Adding that to TODO.